extensions/amp-ad-network-mgid-impl/0.1/amp-ad-network-mgid-impl.js

```diff
@@ -0,0 +1,294 @@
+              'src': pvUrl,
+            })
+          );
+
+        this.win['_mgAmpStoryPV'] = 1;
+      }
+
+      return servicerUrl;
+    });
+  }
+
+  /**
+   * @param {!ConsentTupleDef=} consentTuple
+   * @return {string[]} Consents parameters.
+   * @private
+   */
+  getConsents_(consentTuple) {
+    const result = [];
+
+    let consentState = undefined;
+    let consentString = undefined;
+    let gdprApplies = undefined;
+    let consentStringType = undefined;
+    if (consentTuple) {
+      consentState = consentTuple.consentState;
+      consentString = consentTuple.consentString;
+      gdprApplies = consentTuple.gdprApplies;
+      consentStringType = consentTuple.consentStringType;
+    }
+    if (
+      consentState === CONSENT_POLICY_STATE.UNKNOWN &&
+      this.element.getAttribute('data-npa-on-unknown-consent') !== 'true'
+    ) {
+      return result;
+    }
+
+    if (gdprApplies) {
+      result.push(
+        'gdprApplies=' +
+          (gdprApplies === true ? '1' : gdprApplies === false ? '0' : null)
+      );
+    }
+
+    if (
+      consentString &&
+      consentStringType != CONSENT_STRING_TYPE.US_PRIVACY_STRING
+    ) {
+      result.push('consentData=' + consentString);
+    }
+
+    if (
+      consentString &&
+      consentStringType == CONSENT_STRING_TYPE.US_PRIVACY_STRING
+    ) {
+      result.push('uspString=' + consentString);
+    }
+
+    return result;
+  }
+
+  /**
+   * @return {string} Page data for ad request
+   * @private
+   */
+  getPageParam_() {
+    const counter = Services.urlReplacementsForDoc(
+      this.element
+    )./*OK*/ expandStringSync('COUNTER', undefined, {
+      'COUNTER': true,
+    });
+    return parseInt(counter, 10) % 20;
+  }
+
+  /**
+   * @return {string} Cachebuster for ad request
+   * @private
+   */
+  getCacheBusterParam_() {
+    return (
+      'cbuster=' +
+      Date.now().toString() +
+      Math.floor(Math.random() * 1000000000 + 1)
+    );
+  }
+
+  /**
+   * @return {string} Network information data for ad request
+   * @private
+   */
+  getNetworkInfoParams_() {
+    const params = [];
+    try {
+      const networkInformation =
+        navigator.connection ||
+        navigator.mozConnection ||
+        navigator.webkitConnection;
+
+      if (typeof networkInformation.type != 'undefined') {
+        params.push('nit=' + networkInformation.type);
+      }
+      if (typeof networkInformation.effectiveType != 'undefined') {
+        params.push('niet=' + networkInformation.effectiveType);
+      }
+      if (typeof networkInformation.saveData != 'undefined') {
+        params.push('nisd=' + (networkInformation.saveData ? 1 : 0));
+      }
+    } catch (e) {}
+
+    return params;
+  }
+
+  /**
+   * @return {string} Device pixel ratio info for ad request
+   * @private
+   */
+  getDevicePixelRatioParam_() {
+    let ratio = 1;
+
+    if (typeof window.devicePixelRatio !== 'undefined') {
```

Comment:
is this ever true with `!==`?

Reply:
Yes, it's true for IE<=10, old Opera and FF. I can remove this block if we don't support them.

Reply:
technically we dont support, but we can leave as defense in depth.